packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -164,31 +164,30 @@ export class DefaultCrudRepository<T extends Entity, ID>
    * @param relationName Name of the relation defined on the source model
    * @param targetRepo Target repository instance
    */
-  protected _createHasManyRepositoryFactoryFor<
-    Target extends Entity,
-    TargetID,
-    ForeignKeyType
-  >(
+  protected _createHasManyRepositoryFactoryFor<Target extends Entity, TargetID>(
     relationName: string,
     targetRepo: Getter<EntityCrudRepository<Target, TargetID>>,
-  ): HasManyRepositoryFactory<Target, ForeignKeyType> {
+  ): HasManyRepositoryFactory<Target, ID> {
     const meta = this.entityClass.definition.relations[relationName];
-    return createHasManyRepositoryFactory<Target, TargetID, ForeignKeyType>(
+    return createHasManyRepositoryFactory<Target, TargetID, ID>(
       meta as HasManyDefinition,
       targetRepo,
     );
   }
 
-  protected _createBelongsToFactoryFor<
+  protected _createBelongsToRepositoryFactoryFor<
```

Comment:
Please remove "Repository" from the name of this method, since it does not return any repository.

Proposed name: `_createBelongsToAccessorFor` or `_createBelongsToFactoryFor`.

While we are discussing these names. I find `createBelongsToFactory` name as misleading, because this helper function is not returning a factory (a function creating something new), it returns a function for accessing the related model instance (an accessor, a getter).

Could you please take another look and see if you can come with names that better communicate the intent of these two functions/methods?